args.go

```diff
@@ -6,55 +6,55 @@ var (
 	argsRangeErr = errors.New("index out of range")
 )
 
-// Args wraps a string slice with some convenience methods
-type Args struct {
-	slice []string
-}
-
-// Get returns the nth argument, or else a blank string
-func (a *Args) Get(n int) string {
-	if len(a.slice) > n {
-		return a.slice[n]
+type Args interface {
+	// Get returns the nth argument, or else a blank string
+	Get(n int) string
+	// First returns the first argument, or else a blank string
+	First() string
+	// Tail returns the rest of the arguments (not the first one)
+	// or else an empty string slice
+	Tail() []string
+	// Len returns the length of the wrapped slice
+	Len() int
+	// Present checks if there are any arguments present
+	Present() bool
+	// Slice returns a copy of the internal slice
+	Slice() []string
+}
+
+type args []string
+
+func (a *args) Get(n int) string {
+	if len(*a) > n {
+		return (*a)[n]
 	}
 	return ""
 }
 
-// First returns the first argument, or else a blank string
-func (a *Args) First() string {
+func (a *args) First() string {
 	return a.Get(0)
 }
 
-// Tail returns the rest of the arguments (not the first one)
-// or else an empty string slice
-func (a *Args) Tail() []string {
+func (a *args) Tail() []string {
```

Comment:
Maybe this should call `Slice()` and _slice_ the slice? Would eliminate a little duplication, but honestly this is fine.